### PR TITLE
Convert OOP-based code to ECS

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Convert OOP-based code to ECS (#20)
  - Add library `entt`
  - Convert classes to components using `struct`
  - Refactor related code

### CODE REVIEW

File: Includes/Runeterra/Commons/Constants.hpp [hunk 1]
```diff
@@ -13,6 +13,9 @@ constexpr int NUM_ALL_CARDS = 844;
 //! The max known version.
 const int MAX_KNOWN_VERSION = 2;
 
+//! The number of players.
+const int NUM_PLAYERS = 2;
```
Comment (SeokminHong) [inline Includes/Runeterra/Commons/Constants.hpp:17]: Are there reasons for "MAX_KNOWN_VERSION" and "NUM_PLAYERS" are not "constexpr" variables?

File: Includes/Runeterra/Systems/Engine.hpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    //! \param nameToFind The card name to find.
+    //! \return A card code that matches \p nameToFind.
+    [[nodiscard]] std::optional<std::string> FindCardCodeByName(
+        std::string_view&& nameToFind);
+
+ private:
+    //! Loads a card data from JSON files.
+    void LoadCardData();
+
+    entt::registry m_registry;
```
Comment (SeokminHong) [inline Includes/Runeterra/Systems/Engine.hpp:52]: Are there criteria for naming with Hungarian notations for member variables?

**Review by SeokminHong — COMMENTED** [review]
I added two questions for codes.

File: Includes/Runeterra/Commons/Constants.hpp (hunk 1 above)
Comment (utilForever) [reply Includes/Runeterra/Commons/Constants.hpp:17]: Oh, this is my mistake. I'll fix it! 🧷

**Review by SeokminHong — APPROVED** [review]
LGTM!😉

**Pull request merged**